### ISSUE TITLE
Crash on Disconnect from Vehicle

### ISSUE THREAD

**Issue by DonLakeFlyer** [mavlink/qgroundcontrol#2641]
- Direct connect Vehicle via USB to Android tablet
- Wait for everything to crank up
- Power down Vehicle
- Click Disconnect
- Crash

**Comment by DonLakeFlyer** [comment]
Caused by:
```
SerialLink::~SerialLink()
{
    // Disconnect link from configuration
    _config->setLink(NULL);
    _disconnect();
    if(_port) delete _port;
    _port = NULL;
}
```
The setLink call emits a signal while in the destructor which confuses Qt since some objects are in the middle of going away.

@dogmaphobic Is it necessary to clear the link here since the whole object is going away anyway?

**Comment by dogmaphobic** [comment]
"Fixed" in #3500
